### PR TITLE
bugfix(mailchimp): Adding support for backward compatibility.

### PR DESCRIPTION
**Pull request by ItsSudip**
Adding support for backward compatibility, with adding merge_fields while updating a subscriber. The support for merge_fields is added by an open-source [here](https://github.com/rudderlabs/rudder-transformer/pull/1098/files) is the PR link.

## Description of the change

> Description here

## Type of change
- [X] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [X] The code changed/added as part of this pull request has been covered with tests
- [X] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [X] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/mailchimp/transform.js [hunk 1]
```diff
@@ -137,7 +137,7 @@ async function getPayload(
     Object.keys(traits).forEach(trait => {
       if (trait === "email") {
         rawPayload.email_address = traits[trait];
-      } else {
+      } else if (mailChimpConfig.useUpdatedMapping) {
```
Comment (utsabc) [inline v0/destinations/mailchimp/transform.js:140]:
```suggestion
      } else if (mailChimpConfig.enableMergeFields) {
```

File: v0/destinations/mailchimp/transform.js [hunk 2]
```diff
@@ -214,6 +214,10 @@ function getMailChimpConfig(message, destination) {
       case destinationConfigKeys.dataCenterId:
         mailChimpConfig.dataCenterId = `${destination.Config[key]}`;
         break;
+      case "useUpdatedMapping":
```
Comment (utsabc) [inline v0/destinations/mailchimp/transform.js:217]:
```suggestion
      case "enableMergeFields":
```

File: v0/destinations/mailchimp/transform.js (hunk 1 above)
Comment (ItsSudip) [reply v0/destinations/mailchimp/transform.js]: Changed.

File: v0/destinations/mailchimp/transform.js (hunk 2 above)
Comment (ItsSudip) [reply v0/destinations/mailchimp/transform.js]: Changed

File: __tests__/data/mailchimp_output.json [hunk 3]
```diff
@@ -137,5 +137,28 @@
     },
     "version": "1",
     "endpoint": "https://us20.api.mailchimp.com/3.0/lists/df42a82d07/members/48cd6232dc124497369f59c33d3eb4ab"
+  },
+  {
+    "version": "1",
```
Comment (utsabc) [inline __tests__/data/mailchimp_output.json:142]: Lets add a test case where we have merge fields as output

**Review by utsabc — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**